framework/decode/screenshot_handler.cpp

```diff
@@ -38,10 +39,130 @@ static constexpr size_t kUnormIndex = 0;
 static constexpr size_t kSrgbIndex  = 1;
 
 const VkFormat kImageFormats[][2] = {
-    // Vulkan image formats for ScreenshotFormat::kBmp
-    { VK_FORMAT_B8G8R8A8_UNORM, VK_FORMAT_B8G8R8A8_SRGB }
+    // Vulkan image formats for util::ScreenshotFormat::kBmp
+    { VK_FORMAT_B8G8R8A8_UNORM, VK_FORMAT_B8G8R8A8_SRGB },
+    // Vulkan image formats for util::ScreenshotFormat::kPng
+    { VK_FORMAT_R8G8B8A8_UNORM, VK_FORMAT_R8G8B8A8_SRGB }
 };
 
+inline void OutputImage(const std::string&     filename,
+                        util::ScreenshotFormat file_format,
+                        uint32_t               width,
+                        uint32_t               height,
+                        uint64_t               size,
+                        void*                  data)
+{
+    switch (file_format)
+    {
+        default:
+            GFXRECON_LOG_ERROR("Screenshot format invalid!  Expected BMP or PNG, falling back to BMP.");
+            // Intentional fall-through
+        case util::ScreenshotFormat::kBmp:
+            if (!util::imagewriter::WriteBmpImage(filename + ".bmp", width, height, size, data))
+            {
+                GFXRECON_LOG_ERROR("Screenshot could not be created: failed to write BMP file %s", filename.c_str());
+            }
+            break;
+#ifdef ENABLE_PNG_SCREENSHOT
+        case util::ScreenshotFormat::kPng:
+            // For PNG format, we have to force the alpha channel to opaque, otherwise we get
+            // image artifacts on most image viewers from the transparency for anything that was
+            // rendered last to the image using transparency (think tree leaves).
+            uint32_t* working_data = reinterpret_cast<uint32_t*>(data);
```

Comment:
Bakes in a little-endian assumption. Covers ARM, IA32, x64, RISC-V, etc. Do we care?

Reply:
No, I don't think we care.  If someone wants to capture and replay Vulkan on a Mac SE/30, we'll fix it then.

Reply:
Good catch.  Yes, I think we do since some Android targets may be big-endian.

Reply:
Done.  Added a new `IsBigEndian` check in `framework/util/platform.h` and use that then flip the value based on big-endianness.  I believe it's correct if you want to double check.

Reply:
Actually, I don't see why this couldn't be a `uint8_t*` and just set the appropriate byte to 0xFF.  Also, does this behave the same way as BMP?

Reply:
I think we want to output whatever was in the framebuffer for alpha, so I think you should just omit this loop.

Reply:
(The output is primarily for comparison and regression testing, not for viewing, so it should represent the actual framebuffer.)

Reply:
If you output with alpha enabled on PNG the screen capture format is messed up.  Anywhere that has alpha (like leaves on a tree) result in the PNG having an alpha'd pixel.  That means it blends with the background of the image viewer and doesn't look correct.  This is why I ended up masking it out.

Reply:
![talos_alphamasked_versus_not](https://user-images.githubusercontent.com/16328578/231608545-4874262d-9f55-4f8a-94f4-7e0870549665.png)

Reply:
I could do the loop to modify it per byte, it will just take potentially longer.  But, that should require an endianness check.

Reply:
It's okay if the alpha results in a screenshot that looks messed up.
Like I said, it's not in there for looking at.
It's in there for regression testing, for image comparison with a program.
If a user needs to look at it, they can use a tool that disables alpha.
Setting the alpha to 0xFF removes information from the screenshot.

Why would modifying the data per byte take longer?

Reply:
*also* note that the BMP writer that's already in there *does* write alpha.